File: app/src/main/java/com/uns/uu/uupaymentsdk/InputBankCardActivity.java
package com.uns.uu.uupaymentsdk;

import android.os.Bundle;
import android.support.v7.widget.AppCompatEditText;
import android.support.v7.widget.AppCompatTextView;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;

/**
 * author: 张承
 * time：2018/2/5
 * des：
 */

public class InputBankCardActivity extends BaseActivity implements View.OnClickListener {

    private AppCompatEditText mTvUserName;

    private AppCompatEditText mEtBankCard;

    private AppCompatTextView mTvNext;
    private String name;

    @Override
    protected int getLayout() {
        return R.layout.activity_input_bank_card;
    }

    @Override
    protected void initView() {

        mTvUserName = findViewById(R.id.tv_user_name);
        mEtBankCard = findViewById(R.id.et_bank_card);
        mEtBankCard.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {

            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });
        mTvNext = findViewById(R.id.tv_next);
        mTvNext.setOnClickListener(this);
    }

    @Override
    protected void initData() {
        name = getIntent().getStringExtra("name");
        mTvUserName.setText(getHideName(name));
    }


    @Override
    public void onClick(View v) {
        switch (v.getId()) {

            case R.id.tv_next:
                break;
        }
    }

    private String getHideName(String name) {
        StringBuffer sb = new StringBuffer();
        for (int i = 0; i < name.length() - 1; i++) {
            sb.append("*");
        }
        sb.append(name.substring(name.length() - 1));
        return sb.toString();
    }
}

File: app/src/main/java/com/uns/uu/uupaymentsdk/view/InputBankCardActivity.kt
package com.uns.uu.uupaymentsdk.view

import android.arch.lifecycle.Observer
import android.content.Intent
import android.text.Editable
import android.text.TextWatcher
import com.uns.uu.uupaymentsdk.R
import com.uns.uu.uupaymentsdk.constant.CardBinConstant
import com.uns.uu.uupaymentsdk.utils.HintDialogUtils
import com.uns.uu.uupaymentsdk.viewmodel.GetCardInfoViewModel
import kotlinx.android.synthetic.main.activity_input_bank_card.*

/**
 * author: 张承
 * time：2018/2/23
 * des：输入银行卡号
 */
class InputBankCardActivity : BaseActivity() {

    private var canClick: Boolean = false
    override fun getLayout(): Int {
        return R.layout.activity_input_bank_card
    }

    override fun initView() {
        setTitle("忘记支付密码")
        if (intent.hasExtra("userName")) {
            tv_user_name.text = getHideName(intent.getStringExtra("userName"))
        }
        tv_input_card_next.setOnClickListener {
            if (canClick) {
                GetCardInfoViewModel().getCardInfo(et_input_bank_card.text.toString()).observe(this, Observer {
                    if (CardBinConstant.YES == it?.retCode) {
                        val intent = Intent(baseContext, InputBindCardInfoActivity::class.java)
                        intent.putExtra("cardId", et_input_bank_card.text.toString().trim())
                        intent.putExtra("cardType", "${it.data?.issName}${it.data?.cardTypeName}")
                        startActivity(intent)
                        finish()
                    } else {
                        showTip(it?.retCode + "")
                    }
                })

            }
        }
    }

    override fun initData() {
        et_input_bank_card.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {

            }

            override fun afterTextChanged(s: Editable) {
                if (et_input_bank_card.text.toString().trim().isNotEmpty()) {
                    canClick = true
                    @Suppress("DEPRECATION")
                    tv_input_card_next.background = resources.getDrawable(R.drawable.btn_bg_selected)
                } else {
                    canClick = false
                    @Suppress("DEPRECATION")
                    tv_input_card_next.background = resources.getDrawable(R.drawable.btn_bg_unselect)
                }
            }
        })
    }

    private fun getHideName(name: String): String {
        val sb = StringBuffer()
        for (i in 0 until name.length - 1) {
            sb.append("*")
        }
        sb.append(name.substring(name.length - 1))
        return sb.toString()
    }

    private fun showTip(content: String) {
        val dialog = HintDialogUtils(this)
        dialog.setLeftOrRight(false, "", true, "知道了")
        dialog.setContent(true, content)
        dialog.showDialog()
    }

}